catalog/core/catalog-core-api/src/main/java/ddf/catalog/plugin/PolicyPlugin.java

```diff
@@ -0,0 +1,97 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p/>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p/>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.plugin;
+
+import java.io.Serializable;
+import java.util.List;
+import java.util.Map;
+
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.Result;
+import ddf.catalog.operation.DeleteRequest;
+import ddf.catalog.operation.Query;
+
+/**
+ * A PolicyPlugin is used to build policy information regarding the Catalog action a user is attempting.
+ */
```

Comment:
This may be a good spot to explain how policies are used and applied and should be implemented.